metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -474,50 +472,20 @@ def _create_lineage_map(self, entries: Iterable[QueryEvent]) -> Dict[str, Set[st
                     lineage_map[destination_table_str].add(ref_table_str)
         return lineage_map
 
-    def get_latest_partitions_for_schema(self, schema: str) -> None:
-        query_limit: int = 500
-        offset: int = 0
+    def get_latest_partition(
+        self, schema: str, table: str
+    ) -> Optional[BigQueryPartitionColumn]:
         url = self.config.get_sql_alchemy_url()
         engine = create_engine(url, **self.config.options)
         with engine.connect() as con:
             inspector = inspect(con)
-            partitions = {}
-
-            def get_partition_columns(
-                project_id: str, schema: str, limit: int, offset: int
-            ) -> int:
-                sql = BQ_GET_LATEST_PARTITION_TEMPLATE.format(
-                    project_id=project_id,
-                    schema=schema,
-                    limit=limit,
-                    offset=offset,
-                )
-                result = con.execute(sql)
-                row_count: int = 0
-                for row in result:
-                    partition = BigQueryPartitionColumn(**row)
-                    partitions[partition.table_name] = partition
-                    row_count = row_count + 1
-                return row_count
-
-            res_size = get_partition_columns(
-                self.get_db_name(inspector), schema, query_limit, offset
+            sql = BQ_GET_LATEST_PARTITION_TEMPLATE.format(
+                project_id=self.get_db_name(inspector), schema=schema, table=table
             )
-            while res_size == query_limit:
-                offset = offset + query_limit
-                res_size = get_partition_columns(
-                    self.get_db_name(inspector), schema, query_limit, offset
-                )
-
-            self.partiton_columns[schema] = partitions
-
-    def get_latest_partition(
-        self, schema: str, table: str
-    ) -> Optional[BigQueryPartitionColumn]:
-        if schema not in self.partiton_columns:
-            self.get_latest_partitions_for_schema(schema)
-
-        return self.partiton_columns[schema].get(table)
+            result = con.execute(sql)
+            for row in result:
+                return BigQueryPartitionColumn(**row)
```

Comment:
should this be `yield` of the `BigQueryPartitionColumn`? Otherwise wont this just return the first row?